### DISCUSSION TITLE
How to force close a pooled connection?

### DISCUSSION THREAD

**Discussion by andy-yx-chen** [mysql-net/MySqlConnector#1118] [Q&A]
There is a case that reset connection for XA transactions may bring back a prepared XA transaction which is already rollback by repairer, I don't want all XA related connections are not pooled, but just to force close connections that are running into issues

**Comment by bgrainger** [comment]
The only way is to call [`MySqlConnection.ClearPool`](https://mysqlconnector.net/api/mysqlconnector/mysqlconnection/clearpool/), which closes all pooled connections with the same connection string.

**Reply by andy-yx-chen** [reply]
ClearPool does not work, I eventually figure out that I can do "KILL CONNECTION_ID()" to kill the current connection and turns out that this driver can handle the error from server for killing, thanks for this!
